frontend/src/utils/CachedMergeObservable.ts

```diff
@@ -0,0 +1,50 @@
+import { Observable, OperatorFunction } from 'rxjs';
+import { map } from 'rxjs/operators';
+import CachedObservable, {
+  ExtendedObservableNotification,
+} from './CachedObservable';
+
+export type CachedMergeObservableCache<T> = {
```

Comment:
Using JS object instances for caching will work, but keeps all ever loaded data in the frontend memory. A better approach would be to limit the max number of stored entries with some caching strategy in place, for example LRU. There are existing types that can be used as replacement for JS object instances. See for example: https://github.com/isaacs/node-lru-cache

Reply:
Yes this was the plan. I decided to implement it with a Map for now due to time limitations.